test/extended/testdata/templates/templateinstance_objectkinds.yaml

```diff
@@ -5,7 +5,8 @@ items:
   apiVersion: v1
   metadata:
     name: configsecret
-
+  stringData:
+    FOO: bar
```

Comment:
I think it would be better here if:
a) the secret data actually corresponded to a parameter used by the template (you'll have to invent one)
b) ideally, the describer code listed template parameter values used, rather than all the secret data it finds

Reply:
ideally even taking into account template defaults

Reply:
This isn't really testing that parameters get interpolated correctly (there should already be one for that?), it's just to test the output.

I'm not sure how I'd determine what parameters have been used vs just listing all the parameters defined in the secret, but i'm not listing all of the secret information, just the params.

Reply:
At the moment in DescribeParameters you're ranging over secret.Data; I'd expect you to be ranging over templateinstance.Spec.Template.Parameters, and checking if there's a matching key in secret.Data.  If so, print it.

At the moment the templateinstance controller doesn't persist generated parameters anywhere, which complicates printing non-overriden generated values here.  @bparees do you think this is something that should be dealt with somehow?

Reply:
@jim-minter should it be dealt with somehow? yes.  As part of this effort? no.  Create a card please.

Reply:
https://trello.com/c/kH13stsr/1406-discover-parameters-values-generated-via-templateinstance-api

@coreydaley for now, I think ranging over templateinstance.Spec.Template.Parameters, and checking if there's a matching key in secret.Data, would be a good plan.

Reply:
@jim-minter updated per your suggestion